### PR TITLE
correct empty point submission check for metric submission

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

In https://github.com/anza-xyz/agave/pull/842, the fix that check empty point
is not correct since combined-points alwasy append a 'metric' point at the end. 

#### Summary of Changes

fix empty point check by len() > 1

Fixes #

### CODE REVIEW

**Review by lijunwangs — DISMISSED** [review]
looks good.

File: metrics/src/metrics.rs [hunk 1]
```diff
@@ -280,7 +280,10 @@ impl MetricsAgent {
             counters,
         );
 
-        if !combined_points.is_empty() {
+        // `combined_points`` always contain a `metric` point for how many real
+        // datapoints are submitted at the end. Therefore, to skip empty point
+        // submission, `combined_points.len()` must be larger than 1.
+        if combined_points.len() > 1 {
             writer.write(combined_points);
         }
```
Comment (illia-bobyr) [inline metrics/src/metrics.rs:288]: This is actually something I was not sure about, when I was changing this code.
Is there any value is submitting metrics stats that contain zeroes?

I thought that there is, as it serves as a kind of ping, indicating that metrics are properly uploaded.  There is just nothing to upload.
While if we do not submit anything, it is impossible to distinguish a case when there are no metrics to submit, from a case when something is broken in the metrics upload process.

I should have removed the `if` statement completely, I guess.

But as you are proposing this change, there is probably a flaw or some practical issue with my approach, right?
Could you explain why are you suggesting this change?

File: metrics/src/metrics.rs [hunk 2]
```diff
@@ -683,10 +686,11 @@ mod test {
...
         agent.flush();
 
-        // We are expecting `max_points_per_sec - 1` data points from `submit()` and two more metric
+        // We are expecting `max_points_per_sec - 1` data points from `submit()` and one more metric
         // stats data points.  One from the timeout when all the `submit()`ed values are sent when 1
-        // second is elapsed, and then one more from the explicit `flush()`.
-        assert_eq!(writer.points_written(), max_points_per_sec + 1);
+        // second is elapsed, and the explicit `flush()` won't submit any point due to the fact that
+        // there is no additional points left to submit (empty flush doesn't submit stats data point).
+        assert_eq!(writer.points_written(), max_points_per_sec);
```
Comment (illia-bobyr) [inline metrics/src/metrics.rs:693]: Thanks for updating the comment as well!

File: metrics/src/metrics.rs (hunk 1 above)
Comment (HaoranYi) [reply metrics/src/metrics.rs:288]: This change makes `test_flush_before_drop` fails intermittently on CI. 
More details can be found on discord.
 https://discord.com/channels/428295358100013066/560503042458517505/1232341710885879870

Comment (illia-bobyr) [reply metrics/src/metrics.rs:288]: Sorry for a slow response.
What do you think about this https://github.com/anza-xyz/agave/pull/1039 instead?

**Comment by HaoranYi** [comment]
Closed in favor of #1039

**Pull request closed without merging**